query_optimizer/ExecutionGenerator.cpp

```diff
@@ -1232,8 +1242,11 @@ void ExecutionGenerator::convertInsertSelection(
       execution_plan_->addRelationalOperator(insert_selection_op);
   insert_destination_proto->set_relational_op_index(insert_selection_index);
 
+  CatalogRelation *mutable_relation =
+      catalog_database_->getRelationByIdMutable(selection_relation->getID());
```

Comment:
@jianqiao This is the root cause of my tests failures. We should change to `destination_relation` from `selection_relation`.